app/src/main/java/no/usn/mob3000/data/source/remote/social/FriendsDataSource.kt

```diff
@@ -0,0 +1,19 @@
+package no.usn.mob3000.data.source.remote.social
+
+import io.github.jan.supabase.SupabaseClient
+import io.github.jan.supabase.postgrest.from
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import no.usn.mob3000.data.model.social.FriendSingleDto
+import no.usn.mob3000.data.network.SupabaseClientWrapper
+
+open class FriendsDataSource(
+    private val supabaseClient: SupabaseClient = SupabaseClientWrapper.getClient())
+{
+    suspend fun fetchAllFriends(): List<FriendSingleDto> = withContext(Dispatchers.IO) {
+        supabaseClient
+            .from("friends")
```

Comment:
Remember to look through the existing database code and see if there are RPC functions that you can use. In this case, there are functions to get friends already and may be useful here compared to simply selecting from the friends table.